phpBB/phpbb/style/manager.php

```diff
@@ -0,0 +1,496 @@
+		$this->db->sql_freeresult($result);
+
+		if ($conflict !== false)
+		{
+			throw new exception('STYLE_UNINSTALL_DEPENDENT');
+		}
+
+		// Change default style for users
+		$sql = 'UPDATE ' . $this->users_table . '
+			SET user_style = ' . $this->config['default_style'] . '
+			WHERE user_style = ' . $id;
+
+		if (!$this->db->sql_query($sql))
+		{
+			throw new exception('STYLE_UNINSTALL_UNABLE_UPDATE_USERS');
+		}
+
+		// Uninstall style
+		$sql = 'DELETE FROM ' . $this->styles_table . '
+			WHERE style_id = ' . $id;
+
+		if (!$this->db->sql_query($sql))
+		{
+			throw new exception('STYLE_NOT_UNINSTALLED');
+		}
+	}
+
+	public function activate($ids)
+	{
+		// Check if all the selected styles exist and are deactivated
+		$sql = 'SELECT COUNT(*) as num_styles
+			FROM ' . $this->styles_table . '
+			WHERE style_active = 0 AND ' . $this->db->sql_in_set('style_id', $ids);
+		$result = $this->db->sql_query($sql);
+
+		$nb_styles = (int) $this->db->sql_fetchfield('num_styles');
+
+		$this->db->sql_freeresult($result);
+
+		if ($nb_styles != count($ids))
+		{
+			throw new exception('STYLE_ERROR_ACTIVATE');
+		}
+
+		// Activate styles
+		$sql = 'UPDATE ' . $this->styles_table . '
+			SET style_active = 1
+			WHERE ' . $this->db->sql_in_set('style_id', $ids);
+		$this->db->sql_query($sql);
+
+		// Purge cache
+		$this->cache->destroy('sql', $this->styles_table);
+	}
+
+	public function deactivate($ids)
+	{
+		// Check for default style
+		foreach ($ids as $id)
+		{
+			if ($id == $this->config['default_style'])
+			{
+				throw new exception('DEACTIVATE_DEFAULT');
+			}
+		}
+
+		// Check if all the selected styles exist and are activated
+		$sql = 'SELECT COUNT(*) as num_styles
+			FROM ' . $this->styles_table . '
+			WHERE style_active = 1 AND ' . $this->db->sql_in_set('style_id', $ids);
+		$result = $this->db->sql_query($sql);
+
+		$nb_styles = (int) $this->db->sql_fetchfield('num_styles');
+
+		$this->db->sql_freeresult($result);
+
+		if ($nb_styles != count($ids))
+		{
+			throw new exception('STYLE_ERROR_DEACTIVATE');
+		}
+
+		// Reset default style for users who use selected styles
+		$sql = 'UPDATE ' . $this->users_table . '
+			SET user_style = ' . $this->config['default_style'] . '
+			WHERE ' . $this->db->sql_in_set('style_id', $ids);
+
+		if (!$this->db->sql_query($sql))
+		{
+			throw new exception('STYLE_DEACTIVATE_UNABLE_UPDATE_USERS');
+		}
+
+		// Deactivate styles
+		$sql = 'UPDATE ' . $this->styles_table . '
+			SET style_active = 0
+			WHERE ' . $this->db->sql_in_set('style_id', $ids);
+
+		if (!$this->db->sql_query($sql))
+		{
+			throw new exception('STYLE_ERROR_DEACTIVATE');
+		}
+
+		// Purge cache
+		$this->cache->destroy('sql', $this->styles_table);
+	}
+
+	/**
+	* Delete all files in style directory
+	*
+	* @param string $path Style directory
+	* @param string $dir Directory to remove inside style's directory
+	* @return bool True on success, false on error
+	*/
+	public function delete_style_files($path)
```

Comment:
https://github.com/rubencm/phpbb/blob/ticket/15167/phpBB/includes/acp/acp_styles.php#L304

I can call it from uninstall, but it think it is better like this